plugins/CoreConsole/Commands/DevelopmentSyncUITestScreenshots.php

```diff
@@ -73,42 +71,166 @@ protected function execute(InputInterface $input, OutputInterface $output)
-                Http::sendHttpRequest("$urlBase/processed-ui-screenshots/$file", $timeout = 60, $userAgent = null,
-                    PIWIK_DOCUMENT_ROOT . "/" . $downloadTo);
+                $this->downloadProcessedScreenshot($input, $output, $file);
             }
         }
 
-        $this->displayGitInstructions($output);
+        $this->displayGitInstructions($input, $output);
 
     }
 
-    /**
-     * @param OutputInterface $output
-     */
-    protected function displayGitInstructions(OutputInterface $output)
+    protected function displayGitInstructions(InputInterface $input, OutputInterface $output)
     {
         $output->writeln('');
         $output->writeln('--------------');
         $output->writeln('');
         $output->writeln("If all downloaded screenshots are valid you may push them with these commands:");
-        $output->writeln('');
-        $commands = "cd tests/UI/expected-ui-screenshots
+        $downloadToPath = $this->getDownloadToPath($input);
+        $commands = "
+cd $downloadToPath
 git pull
 git add .
-git commit -m '' # WRITE A COMMIT MESSAGE
-git push
+git commit -m '' # Write a good commit message
+git push";
+
+        $plugin = $input->getOption('plugin');
+        if(empty($plugin)) {
+            $commands .= "
 cd ..
 git pull
-git add expected-ui-screenshots
-git commit -m '' #WRITE A COMMIT MESSAGE
-git push";
+git add expected-ui-screenshots/
+git commit -m '' # Copy paste the good commit message
+git push
+cd ../../\n\n";
+        } else {
+            $commands .= "
+cd ../../../../../\n\n";
+        }
         $output->writeln($commands);
     }
+
+    protected function getUrlBase(InputInterface $input)
+    {
+        $buildNumber = $input->getArgument('buildnumber');
+        if (empty($buildNumber)) {
+            throw new \InvalidArgumentException('Missing build number.');
+        }
+
+        $plugin = $input->getOption('plugin');
+        if ($plugin) {
+            $urlBase = sprintf('http://builds-artifacts.piwik.org/ui-tests.master.%s/%s', $plugin, $buildNumber);
+        } else {
+            $urlBase = sprintf('http://builds-artifacts.piwik.org/ui-tests.master/%s', $buildNumber);
+        }
+        return $urlBase;
+    }
+
+    protected function getDownloadToPath(InputInterface $input)
+    {
+        $plugin = $input->getOption('plugin');
+
+        $downloadTo = PIWIK_DOCUMENT_ROOT . "/";
+        if (empty($plugin)) {
+            $downloadTo .= "tests/UI/expected-ui-screenshots/";
+        } else {
+            $downloadTo .= "plugins/$plugin/tests/UI/expected-ui-screenshots/";
+        }
+        if(is_dir($downloadTo)) {
+            return $downloadTo;
+        }
+        $downloadTo = str_replace("tests/", "Test/", $downloadTo);
+        if(is_dir($downloadTo)) {
+            return $downloadTo;
+        }
+
+        throw new \Exception("Download to path could not be found: $downloadTo");
+    }
```

Comment:
The logic for this method looks kind of confusing to me. You might find it useful to organize logic by the type of path that is returned, eg, normal UI tests repo path or plugin path.

It might look like:

```
if (empty($plugin)) {
   return $this->getPathToCoreUIExpectedScreens();
} else {
   $possiblePluginPaths = $this->getPossiblePathsToPluginExpectedScreens();

   // loop through possible paths and find one that exists
}
```